ckanext/ckanext-apicatalog_ui/ckanext/apicatalog_ui/views/useradd.py

```diff
@@ -0,0 +1,102 @@
+import random
+import string
+
+import ckan.plugins as p
+
+from flask import Blueprint
+from flask.views import MethodView
+from ckan.model.user import User
+from ckan.lib import mailer
+from logging import getLogger
+from ckan.common import g, _
+from ckan import logic
+
+log = getLogger(__name__)
+useradd = Blueprint('admin_useradd', __name__, url_prefix=u'/ckan-admin')
+
+
+def _generate_random_password():
+    # From ckan.logic.action.create.user_invite
+    # Choose a password. However it will not be used - the invitee will not be
+    # told it - they will need to reset it
+    while True:
+        password = ''.join(random.SystemRandom().choice(
+            string.ascii_lowercase + string.ascii_uppercase + string.digits)
+            for _ in range(12))
+        # Occasionally it won't meet the constraints, so check
+        errors = {}
+        logic.validators.user_password_validator(
+            'password', {'password': password}, errors, None)
+        if not errors:
+            return password
+
+
+@useradd.route('/useradd', endpoint='read', methods=['GET', 'POST'])
```

Comment:
Should this be done with add_url_rules https://github.com/ckan/ckan/wiki/Migration-from-Pylons-to-Flask#flask-views-blueprints-and-routing just to follow common conventions?

Reply:
[The guide](https://github.com/ckan/ckan/wiki/Migration-from-Pylons-to-Flask#flask-views-blueprints-and-routing) specifies both as ok.